### ISSUE TITLE
Typescript error between polaris 10.0.0 and polaris-icons 5.4.0

### ISSUE THREAD

**Issue by RickyMarou** [Shopify/polaris-react-archive#7020]
## Issue summary

`@shopify/polaris` 10.0.0 and `@shopify/polaris-icons` 5.4.0 have type incompatibilities.
## Expected behavior

When using the latest stable version of `@shopify/polaris` (10.0.0) and `@shopify/polaris-icons` (5.4.0), I expect to be able to add an icon from polaris-icons to the `<Icon />` component using the `source` prop without having to deal with a type error from typescript.

## Actual behavior

The following code:

```ts
import { Icon } from "@shopify/polaris";
import { AddMajor } from "@shopify/polaris-icons";

const Test = () => {
  return <Icon source={AddMajor} />;
}
```

will throw the following typescript error:

```
Type 'FunctionComponent<SVGProps<SVGSVGElement>>' is not assignable to type 'IconSource'.
  Type 'React.FunctionComponent<React.SVGProps<SVGSVGElement>>' is not assignable to type 'import("/Users/marwan/random-repos/polaris-icons-debug/node_modules/@shopify/polaris/node_modules/@types/react/index").FunctionComponent<import("/Users/marwan/random-repos/polaris-icons-debug/node_modules/@shopify/polaris/node_modules/@types/react/index").SVGProps<SVGSVGElement>>'.
    Types of parameters 'props' and 'props' are incompatible.
      Type 'PropsWithChildren<SVGProps<SVGSVGElement>>' is not assignable to type 'SVGProps<SVGSVGElement>'.
        Types of property 'children' are incompatible.
          Type 'import("/Users/marwan/random-repos/polaris-icons-debug/node_modules/@shopify/polaris/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'
```

## Steps to reproduce the problem

1. Install `@shopify/polaris` 10.0.0 and `@shopify/polaris-icons` 5.4.0
1. Import an icon and use it with the icon component

## Reduced test case

I created a repository with the error that can be found here: https://github.com/RickyMarou/polaris-icon-debug

Sadly, codesandbox or stackblitz are not showing this error, however I am able to reproduce it in different projects.

## Specifications

- Are you using the React components? Y
- Polaris version number: 10.0.0
- Browser: any
- Device: any
- Operating System: macos 12.5.1

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash
System:
    OS: macOS 12.5.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 111.45 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.0.0 - /opt/homebrew/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.6.0 - /opt/homebrew/bin/npm
    Watchman: 2022.07.04.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 104.0.5112.101
    Firefox: 103.0.1
    Safari: 15.6.1
  npmPackages:
    @shopify/polaris: ^10.0.0 => 10.0.0 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
```

**Comment by aaronccasanova** [comment]
Hey @RickyMarou 👋

I pulled down the repo and noticed "incorrect peer dependency" warnings on `yarn install`:
![image](https://user-images.githubusercontent.com/32409546/186714138-1df6090a-63ba-4e1f-a15e-23e9395ae35b.png)

Updating the `react`, `react-dom`, `@types/react`, and `@types/react-dom` versions to `^17.0.0` resolved the issue for me!

**Comment by BPScott** [comment]
Downgrading react from v18 to v17 should not be a solution. Polaris should support v18 in its peerDependency range (because y'know we're trying to get the admin on v18 too)


This seems like a conflict in `@types/react` between v17 and v18. Deduplicating `@types/react` should be enough, however for some weird reason the yarn.lock seems to think that [polaris depend on @types/react@^17`](https://github.com/RickyMarou/polaris-icon-debug/blob/main/yarn.lock#L312-L313), when it [should allow for v18](https://unpkg.com/browse/@shopify/polaris@10.0.0/package.json#L67).


This seems like the same root issue as #6991

**Comment by RickyMarou** [comment]
@BPScott @aaronccasanova thank you folks for looking into this.

> Downgrading react from v18 to v17 should not be a solution.

+1 on this one, we want to use react 18, not 17.

> yarn.lock seems to think that [polaris depend on @types/react@^17`](https://github.com/RickyMarou/polaris-icon-debug/blob/main/yarn.lock#L312-L313), when it [should allow for v18](https://unpkg.com/browse/@shopify/polaris@10.0.0/package.json#L67).

Are you sure about this? `polaris-react` has its [react peer dependency locked in to 17.x](https://github.com/Shopify/polaris/blob/main/polaris-react/package.json#L73-L76) , and the [dev deps for types are also locked in on 17.x ](https://github.com/Shopify/polaris/blob/main/polaris-react/package.json#L92-L93)

So my understanding is that the core issue is lack of support for react 18, not an incompatibility between two packages from the same monorepo.

**Comment by RickyMarou** [comment]
This has been fixed somewhere between 10.0.0 and 10.8.0
